File: 2070-most-beautiful-item-for-each-query/2070-most-beautiful-item-for-each-query.cpp
class Solution {
public:
    vector<int> maximumBeauty(vector<vector<int>>& item, vector<int>& queries) {
        unordered_map<int,int> map;
        sort(item.begin(),item.end());
        int maxi = -1e9;
        for(auto&x:item) {
            maxi=max(maxi,x[1]);
            map[x[0]]=maxi;
        }
        vector<int> ans;
        for(auto&x:queries) {
            if(map.find(x)==map.end()) {
                int l = 0,r=item.size()-1,ansIdx=-1;
                while(l<=r){
                    int mid = (l+r)>>1;
                    if(item[mid][0] < x) {
                        ansIdx=mid;
                        l=mid+1;
                    } else {
                        r=mid-1;
                    }
                }
                ans.push_back(((ansIdx==-1)?0:map[item[ansIdx][0]]));
            } else {
              ans.push_back(map[x]);
            }
        }
        return ans;
    }
};

File: 2070-most-beautiful-item-for-each-query/2070-most-beautiful-item-for-each-query.py
class Solution:
    def maximumBeauty(self, items: List[List[int]], queries: List[int]) -> List[int]:
        sorted_items = sorted(items, key = lambda x: x[1], reverse = True)
        ans = []
        for query in queries:
            for item in sorted_items:
                if item[0] <= query:
                    ans.append(item[1])
                    break
            else:
                ans.append(0)
        return ans
       